packages/core/src/checks/perform-bulk-return-type.js

```diff
@@ -0,0 +1,60 @@
+const _ = require('lodash');
+
+const performBulkEchoesIds = {
+  name: 'performBulkReturnType',
+
+  shouldRun: (method, bundle) => {
+    return (
+      Array.isArray(bundle.bulk) &&
+      method.endsWith('.operation.performBulk') &&
+      method.startsWith('creates.')
+    );
+  },
+
+  run: (method, results, compiledApp, bundle) => {
+    if (!_.isPlainObject(results)) {
+      // create-is-object should have caught this
+      return [];
+    }
+
+    const inputIds = bundle.bulk
+      .map((b) => {
+        return b && b.meta ? b.meta.id : null;
+      })
+      .filter((id) => id);
+
+    const outputIds = Object.keys(results);
+    const missingIds = inputIds.filter((id) => !outputIds.includes(id));
```

Comment:
@FokkeZB do you think if the output should contain _all_ the IDs in the input (`bundle.bulk`)? The current implementation assumes yes.

Reply:
Updated the requirements in the description. So now each item must have an `outputData` object _or_ an `error` string.